### PR TITLE
Issue 6 implement user authentication using jwt

### PR DESCRIPTION
**Pull request by Hamza029**

### CODE REVIEW

File: src/controllers/authController.ts [hunk 1]
```diff
@@ -19,7 +23,14 @@ const signup = async (
 
     const newUser: IUser = await authService.signup(userInput);
 
-    sendResponse(req, res, 201, 'created', 'successfully signed up', newUser);
+    sendResponse<IUser>(
+      req,
+      res,
+      201,
+      'created',
```
Comment (Propo41) [inline src/controllers/authController.ts:30]: unnecessary field. can remove it if you want

File: src/controllers/userController.ts [hunk 2]
```diff
@@ -11,7 +12,17 @@ export const getAllUsers = async (
...
   try {
     const users: IUser[] = await userService.getAllUsers();
-    sendResponse(req, res, 200, 'success', 'fetched all users', users);
+    sendResponse<IUser[]>(req, res, 200, 'success', 'fetched all users', users);
+  } catch (err) {
+    next(err);
+  }
+};
+
+const protect = async (req: Request, res: Response, next: NextFunction) => {
```
Comment (Propo41) [inline src/controllers/userController.ts:21]: unused method `protect`. where is it used?

File: src/services/userService.ts [hunk 3]
```diff
@@ -11,6 +12,16 @@ const getAllUsers = async (): Promise<IUser[]> => {
   return users;
 };
 
+const protect = async (id: number, token: string | undefined) => {
```
Comment (Propo41) [inline src/services/userService.ts:15]: i assume this is where you are implementing your route protection (authorization). This should be in a middleware.

File: src/interfaces/user.ts [hunk 4]
```diff
@@ -4,7 +4,7 @@ interface IUserAttributes {
   Email: string;
   Name: string;
   JoinDate: Date;
-  Role: number;
+  Role: 0 | 1;
```
Comment (Propo41) [inline src/interfaces/user.ts:7]: better to use an enum `RoleTypes`, ie: 
```ts
enum RoleTypes { 
  USER = 0,
  ADMIN = 1
}
```

**Review by Propo41 — COMMENTED** [review]
..

**Review by edge555 — APPROVED** [review]

**Pull request merged**